### PR TITLE
Fixed flaky tests in ExceptionsTest.java with ObjectMapper

### PR DESCRIPTION
**Pull request by mumbler6**
Hello, I apologize for not creating and linking a JIRA issue for this PR yet, though I did request for a JIRA account on the self-serve page if needed. This PR is fairly trivial though and does not change much code. 

### Purpose of this PR
This PR fixes the flaky tests:
* [`org.apache.servicecomb.core.exception.ExceptionsTest.should_convert_unknown_client_exception_to_invocation_exception`](https://github.com/apache/servicecomb-java-chassis/blob/fc7c01668fbb992e015458a86bc4366a00dc5878/core/src/test/java/org/apache/servicecomb/core/exception/ExceptionsTest.java#L46)
* [`org.apache.servicecomb.core.exception.ExceptionsTest.should_convert_unknown_server_exception_to_invocation_exception`](https://github.com/apache/servicecomb-java-chassis/blob/fc7c01668fbb992e015458a86bc4366a00dc5878/core/src/test/java/org/apache/servicecomb/core/exception/ExceptionsTest.java#L59)

The mentioned tests may non-deterministically pass or fail without changes made to the source code when it is run in different JVMs due to the inherent unordered nature of JSON properties in the JSON strings during comparison. The tests expect: 

* `"{\"code\":\"SCB.00000000\",\"message\":\"Unexpected exception when processing none. msg\"}"`
* `"{\"code\":\"SCB.50000000\",\"message\":\"Unexpected exception when processing none. msg\"}"`

But `Json.encode(invocationException.getErrorData())` will sometimes return: 

* `"{\"message\":\"Unexpected exception when processing none. msg\",\"code\":\"SCB.00000000\"}"`
* `"{\"message\":\"Unexpected exception when processing none. msg\",\"code\":\"SCB.50000000\"}"`

### Steps to Reproduce

This change was confirmed by running the [NonDex](https://github.com/TestingResearchIllinois/NonDex) tool, which explores and reports errors in different behaviors of under-determined Java APIs.
```
# Compile the module in project directory
mvn clean install -pl core -am -DskipTests

# Run the unit tests using NonDex
mvn edu.illinois:nondex-maven-plugin:2.1.7:nondex -pl core -Dtest=org.apache.servicecomb.core.exception.ExceptionsTest#should_convert_unknown_client_exception_to_invocation_exception \
mvn edu.illinois:nondex-maven-plugin:2.1.7:nondex -pl core -Dtest=org.apache.servicecomb.core.exception.ExceptionsTest#should_convert_unknown_server_exception_to_invocation_exception
```
* Following the above steps for the mentioned tests will produce the following results for those runs, respectively:
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ExceptionsTest.should_convert_unknown_client_exception_to_invocation_exception:55 
expected: "{"code":"SCB.00000000","message":"Unexpected exception when processing none. msg"}"
 but was: "{"message":"Unexpected exception when processing none. msg","code":"SCB.00000000"}"
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
```
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ExceptionsTest.should_convert_unknown_server_exception_to_invocation_exception:68 
expected: "{"code":"SCB.50000000","message":"Unexpected exception when processing none. msg"}"
 but was: "{"message":"Unexpected exception when processing none. msg","code":"SCB.50000000"}"
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
```

### Fix
* I propose to use an ObjectMapper to convert the JSON strings into JsonNode trees which can be compared for equality regardless of the order. 
* After the fix, I was able to get both test cases to pass on NonDex: 

image

Please let me know if there are any changes you would like me to make.

### CODE REVIEW

**Comment by shoothzj** [comment]
Thanks for your contribution, I have approve your jira request :)

**Review by shoothzj — APPROVED** [review]
LGTM. In json ouput, I think we don't need to guarante the order

**Comment by mumbler6** [comment]
Hi, thank you! Should I still make an issue on JIRA?

File: core/src/test/java/org/apache/servicecomb/core/exception/ExceptionsTest.java [hunk 1]
```diff
@@ -64,8 +72,15 @@ void should_convert_unknown_server_exception_to_invocation_exception() {
     assertThat(invocationException).hasCause(exception);
     assertThat(invocationException.getStatus()).isEqualTo(INTERNAL_SERVER_ERROR);
     assertThat(invocationException.getErrorData()).isInstanceOf(CommonExceptionData.class);
-    assertThat(Json.encode(invocationException.getErrorData()))
-        .isEqualTo("{\"code\":\"SCB.50000000\",\"message\":\"Unexpected exception when processing none. msg\"}");
+
+    ObjectMapper objectMapper = new ObjectMapper();
```
Comment (caimo) [inline core/src/test/java/org/apache/servicecomb/core/exception/ExceptionsTest.java:76]: objectMapper no need create in each test case.

**Review by caimo — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/apache/servicecomb/core/exception/ExceptionsTest.java (hunk 1 above)
Comment (mumbler6) [reply core/src/test/java/org/apache/servicecomb/core/exception/ExceptionsTest.java:76]: Should I create a static variable available to the entire test class like this?

image

Comment (caimo) [reply core/src/test/java/org/apache/servicecomb/core/exception/ExceptionsTest.java:76]: yes, this will be reuseable

**Pull request merged**